types/flexslider/index.d.ts

```diff
@@ -1,91 +1,390 @@
+         */
+        pausePlay?: boolean;
+        /**
+         * Set the text for the "pause" pausePlay item
+         * @default Pause
+         */
+        pauseText?: string;
+        /**
+         * Set the text for the "play" pausePlay item
+         * @default Play
+         */
+        playText?: string;
 
+        // Special properties
+        /**
+         * Declare which container the navigation elements should be appended too.
+         * Default container is the FlexSlider element. Example use would be
+         * $(".flexslider-container"). Property is ignored if given element is not
+         * found.
+         */
+        controlsContainer?: JQuery;
+        /**
+         * Declare custom control navigation. Examples would be $(".flex-control-nav
+         * li") or "#tabs-nav li img", etc. The number of elements in your
+         * controlNav should match the number of slides/tabs.
+         */
+        manualControls?: JQuery;
+        /**
+         * Custom prev / next button. Must be two jQuery elements. In order to make
+         * the events work they have to have the classes "prev" and "next" (plus
+         * namespace)
+         */
+        customDirectionNav?: JQuery;
+        /**
+         * Mirror the actions performed on this slider with another
+         * slider. Use with care.
+         */
+        sync?: string;
+        /**
+         * Internal property exposed for turning the slider into a
+         * thumbnail navigation for another slider
+         */
+        asNavFor?: string;
+
+        // Carousel Options
+        /**
+         * Box-model width of individual carousel items, including
+         * horizontal borders and padding.
+         * @default 0
+         */
+        itemWidth?: number;
+        /**
+         * Margin between carousel items.
+         * @default 0
+         */
+        itemMargin?: number;
+        /**
+         * Minimum number of carousel items that should be visible.
+         * Items will resize fluidly when below this.
+         * @default 1
+         */
+        minItems?: number;
+        /**
+         * Maxmimum number of carousel items that should be visible.
+         * Items will resize fluidly when above this limit.
+         * @default 0
+         */
+        maxItems?: number;
+        /**
+         * Number of carousel items that should move on animation. If
+         * 0, slider will move all visible items.
+         * @default 0
+         */
+        move?: number;
+        /**
+         * Whether or not to allow a slider comprised of a single slide
+         * @default true
+         */
+        allowOneSlide?: boolean;
+
+        // Browser Specific
+        /**
+         * Set to true when Firefox is the browser used.
+         * @default false
+         */
+        isFirefox?: boolean;
+
+        /**
+         * Whether or not to enable RTL mode
+         * @default false
+         */
+        rtl?: boolean;
+    }
+
+    interface Methods {
```

Comment:
I don't things any of those methods are `undefined` at all.

`$.extend({}, $.flexslider.defaults, options);` will not cause override of defaults

```javascript
 // Callback API
    start: function(){},            //Callback: function(slider) - Fires when the slider loads the first slide
    before: function(){},           //Callback: function(slider) - Fires asynchronously with each slider animation
    after: function(){},            //Callback: function(slider) - Fires after each slider animation completes
    end: function(){},              //Callback: function(slider) - Fires when the slider reaches the last slide (asynchronous)
    added: function(){},            //{NEW} Callback: function(slider) - Fires after a slide is added
    removed: function(){},           //{NEW} Callback: function(slider) - Fires after a slide is removed
    init: function() {},             //{NEW} Callback: function(slider) - Fires after the slider is initially setup
  rtl: false             //{NEW} Boolean: Whether or not to enable RTL mode
  };
```

Reply:
oh I see it now, it's the definition for options only